boards/BMS/Test/Src/Test_StateMachine.cpp

```diff
@@ -628,4 +628,120 @@ TEST_F(BmsStateMachineTest, check_state_transition_from_fault_to_init_with_air_n
     ASSERT_EQ(CANMSGS_BMS_STATE_MACHINE_STATE_INIT_CHOICE, App_CanTx_GetPeriodicSignal_STATE(can_tx_interface));
 }
 
-} // namespace StateMachineTest
+TEST_F(BmsStateMachineTest, check_state_transition_to_fault_state_from_all_states_undervoltage)
+{
+    SetInitialState(App_GetDriveState());
+
+    get_min_cell_voltage_fake.return_val = 2.9f;
+
+    LetTimePass(state_machine, 10);
+    ASSERT_EQ(App_GetFaultState(), App_SharedStateMachine_GetCurrentState(state_machine));
+}
+
+TEST_F(BmsStateMachineTest, check_state_transition_to_fault_state_from_all_states_overvoltage)
+{
+    SetInitialState(App_GetInitState());
+
+    get_max_cell_voltage_fake.return_val = 4.3f;
```

Comment:
just wondering if it's bad practice to put a constant here from the .h file. In the case where we change cell chemistry this could prevent a lot of rework. Thoughts?

Reply:
Personally I'm okay with it, I don't see why it would cause an issue